feat(.NET): Added `EventProcessor` (#7599)
getsentry/sentry-docs: src/platform-includes/enriching-events/event-processors/dotnet.mdx

You can create your own event processors by implementing the `ISentryEventProcessor` interfact. You can also create custom processors for transactions by implementing the `ISentryTransactionProcessor`.

```csharp
using Sentry;
using Sentry.Extensibility;

public class CustomEventProcessor : ISentryEventProcessor
{
    public SentryEvent? Process(SentryEvent @event)
    {
        // Add anything to the event here
        // returning `null` will drop the event
        return @event;
    }
}
```

> There are multie ways of adding the event processors. Processors run on every event sent after they were added.
> Adding it to the options ensures the processor runs with every event after initialization

Reviewer: ```suggestion
Processors run on every event sent once they've been added. While there are multiple ways of doing this, adding processors to the options ensures that they run with every event after initialization.
```

```csharp
options.AddEventProcessor(new CustomEventProcessor());
```

> After adding it to the scope the processor applies to the current and following scopes

Reviewer: ```suggestion
If `AddEventProcessor` has been added to the scope, it'll apply to both the current and the following scopes:
```
